### PR TITLE
Read Feature

### PR DESCRIPTION
**Pull request by yanevdw**
This PR revolves mainly around the implementation which allows users to read a comic and its chapters. Users can select the chapter they wish to read, and navigate to the previous, next and back to all the chapters when they reach the end of a chapter.

**Kindly take note**
- In `_root.tsx`, I currentlly check via the pathname if I am on the chapter page to edit the display for the chapter page (remove footer and styling that affects continuous scrolling). This is currently a temporary solution until I can find a better one.

The PR includes:
- Adding routes, links to navigate to routes and updating the routing configuration to handle reading functionality.
- Adding the `ComicContent` page which displays when the user selects "Read Comic" from the popup. 
    - This shows a list of all the chapters for the comic.
- Adding the `ComicChapter` page to which the user is directed to when the user selects a specific chapter to read.
- Updates to type and file names to remove unused values and reduce null types and increase understandability of component use.

### CODE REVIEW

**Review by tanadendere — APPROVED** [review]
Nice, it is looking good. Well done!

File: src/routes/comic.$comicSlug.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createFileRoute } from "@tanstack/react-router";
+import ComicContent from "../components/comic/ComicContent";
+
+export const Route = createFileRoute("/comic/$comicSlug")({
+  component: Comic,
+});
+
+function Comic() {
+  const { comicSlug } = Route.useParams();
```
Comment (Meggie-H) [inline src/routes/comic.$comicSlug.tsx:9]: Just check that comicSlug's type is not any, it looks to me that it is. But I might be wrong.

File: src/services/api.ts [hunk 2]
```diff
@@ -27,10 +27,11 @@ export async function fetchComicContent(
 
 // Call the API to get a specified comic's chapters based on the comic's hid.
 export async function fetchComicChapters(
-  comicHid: string
+  comicHid: string,
+  chapterCount: number
 ): Promise<ComicChapters> {
   const response = await axios.get(
```
Comment (Meggie-H) [inline src/services/api.ts:33]: Also your response here is not typed

**Review by Meggie-H — COMMENTED** [review]
I would suggest you run your linter and just check for all the implicit any's. This will just make your code safer. 
Other than that it looks good to me. Well done

**Review by Meggie-H — APPROVED** [review]

File: src/components/comic/ComicChapter.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    !(
+      chapterData?.chapter ||
+      chapterData?.chapter?.md_images ||
+      chapterData?.prev ||
+      chapterData?.prev?.hid
+    )
+  ) {
+    return (
+      <p className="p-2 font-sans text-white text-center self-center">
+        This chapter does not contain any content.
```
Comment (LesediMoko) [inline src/components/comic/ComicChapter.tsx:33]: Great use of null checks and letting the user know of missing information 👏

File: src/components/comic/ComicChapter.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            className="bg-blue-munsell w-fit h-fit px-2 py-2 rounded-md text-white font-semibold"
+            onClick={() => (
+              <ComicChapter chapterId={`${chapterData?.next?.hid}`} />
+            )}
+          >
+            Next chapter
+          </Link>
+        ) : null}
+        <Link
+          to={"../../"}
```
Comment (LesediMoko) [inline src/components/comic/ComicChapter.tsx:70]: I'm assuming your are using the location ` ../../ ` to go back to pervious pages. Consider accomplishing the same logic using the useNavigate's ` navigate(-1) ` function. It is part of react-router-dom

Comment (LesediMoko) [reply src/components/comic/ComicChapter.tsx:70]: Just realised you are using tanstack router, never mind 😅 my bad

File: src/services/api.ts [hunk 5]
```diff
@@ -27,10 +27,11 @@ export async function fetchComicContent(
...
-    `${baseURL}comic/${comicHid}/chapters?lang=en`
+    `${baseURL}comic/${comicHid}/chapters?lang=en&limit=${chapterCount * 6}`
```
Comment (LesediMoko) [inline src/services/api.ts:34]: Great use of a large upper-limit in case their are multiple scans for one chapter 👏

**Review by LesediMoko — APPROVED** [review]
I couldn't find much wrong with this pull request. It all looks good! Fantastic work, Yané! 🏄🏽‍♂️

File: src/routes/comic.$comicSlug.tsx (hunk 1 above)
Comment (yanevdw) [reply src/routes/comic.$comicSlug.tsx:9]: Thank you. I will check everything in my next branch.

File: src/services/api.ts (hunk 2 above)
Comment (yanevdw) [reply src/services/api.ts:33]: I will check everything in my improvements and fixes branch.

**Pull request merged**